### PR TITLE
NAS-117407 / 22.12 / Clean up `pool.dataset.query` calls in the new `Datasets` module

### PR DESCRIPTION
**Pull request by denysbutenko**
Details in [the ticket](https://ixsystems.atlassian.net/browse/NAS-117407)

For testing, go to the Datasets page and try to break it

### CODE REVIEW

**Comment by denysbutenko** [comment]
It requires some extra cleanup and fixing unit tests

**Comment by denysbutenko** [comment]
> It requires some extra cleanup and fixing unit tests

Done

File: src/app/pages/datasets/components/dataset-management/dataset-management.component.ts [hunk 1]
```diff
@@ -20,7 +22,20 @@ import { WebSocketService } from 'app/services';
...
+      this.cdr.markForCheck();
+    }),
+    switchMap((dataset) => this.ws.call('pool.dataset.query', [[['id', '=', dataset.id]]])),
+    map((datasets) => datasets[0]),
+    tap(() => {
+      this.isLoadingDataset = false;
+      this.cdr.markForCheck();
+    }),
+  );
   selectedParentDataset$ = this.datasetStore.selectedParentDataset$;
```
Comment (undsoft) [inline src/app/pages/datasets/components/dataset-management/dataset-management.component.ts:39]: This line is unused, but it was used before.

File: src/app/pages/datasets/components/dataset-management/dataset-management.component.ts [hunk 2]
```diff
@@ -20,7 +22,20 @@ import { WebSocketService } from 'app/services';
 })
 export class DatasetsManagementComponent implements OnInit {
   isLoading$ = this.datasetStore.isLoading$;
-  selectedDataset$ = this.datasetStore.selectedDataset$;
+  isLoadingDataset = false;
+  selectedDataset$ = this.datasetStore.selectedDataset$.pipe(
```
Comment (undsoft) [inline src/app/pages/datasets/components/dataset-management/dataset-management.component.ts:26]: This should be part of the state, more specifically an effect with additional properties and selectors.

**Review by undsoft — CHANGES_REQUESTED** [review]
Don't see any obvious regressions, looks good.

File: src/app/pages/datasets/components/dataset-management/dataset-management.component.ts (hunk 1 above)
Comment (denysbutenko) [reply src/app/pages/datasets/components/dataset-management/dataset-management.component.ts:39]: Yes, I'll remove it. Before `selectedParentDataset` was passed from `Dataset Management` to `Dataset Details Panel` and then to `ZFS Encryption Card`. Now it is loaded in `Dataset Details Panel` and then passed to `ZFS Encryption Card`

**Comment by denysbutenko** [comment]
I fixed feedback items. There is [a PR](https://github.com/truenas/middleware/pull/9601) to add more details. So more updates are coming

**Comment by denysbutenko** [comment]
One more middleware PR - https://github.com/truenas/middleware/pull/9603

**Comment by denysbutenko** [comment]
An easy way to test it:
```
Go to TrueNAS UI → Shell.
cd /usr/lib/python3/dist-packages/middlewared/plugins/pool_
rm -rf dataset_details.py
wget https://github.com/truenas/middleware/raw/f9f0dbe1b26ef642af01a6abd97fe2f805020f33/src/middlewared/middlewared/plugins/pool_/dataset_details.py
systemctl restart middlewared
```

**Pull request merged**